### PR TITLE
adds a balloon alert to hulks trying to cast heretic cloak of shadows

### PR DESCRIPTION
**Pull request by spiders24110**
## About The Pull Request
does that
## Why It's Good For The Game
hulk very confused
## Changelog
:cl:
qol: added a balloon alert to hulks trying to use cloak of shadows (they can't use it)
/:cl:

### CODE REVIEW

**Comment by ZephyrTFA** [comment]
> this guy probably knows code more than I do

Nah, its just that we only want the balloon alert showing if the person trying to use it is hulked, we don't want it to always show up regardless

File: code/modules/antagonists/heretic/magic/shadow_cloak.dm [hunk 1]
```diff
@@ -28,7 +28,10 @@
 	return ..()
 
 /datum/action/cooldown/spell/shadow_cloak/is_valid_target(atom/cast_on)
-	return isliving(cast_on) && !HAS_TRAIT(cast_on, TRAIT_HULK) // Hulks are not stealthy. Need not apply
+	if(HAS_TRAIT(cast_on, TRAIT_HULK))
+		cast_on.balloon_alert(cast_on, "cannot cast while hulk")
```
Comment (Fikou) [inline code/modules/antagonists/heretic/magic/shadow_cloak.dm:32]: this is a proc to check whether or not tis a valid target, it should not apply the balloon itself i think?

Comment (ZephyrTFA) [reply code/modules/antagonists/heretic/magic/shadow_cloak.dm:32]: returning false will instantly cancel the spell chain iirc

Comment (ZephyrTFA) [reply code/modules/antagonists/heretic/magic/shadow_cloak.dm:32]: I'm not at my desktop or I could look at the actual call chain

Comment (dragomagol) [inline code/modules/antagonists/heretic/magic/shadow_cloak.dm:32]:
```suggestion
		cast_on.balloon_alert(cast_on, "cannot cast while hulk!")
```

Comment (dragomagol) [reply code/modules/antagonists/heretic/magic/shadow_cloak.dm:32]: Balloon alerts for failures should end with an exclamation

**Pull request merged**